cogs/moderation.py

```diff
@@ -184,37 +145,24 @@ async def duration(self, ctx, id, *, _duration):
 
     @commands.command()
     @commands.has_role("Moderator")
-    async def clean(self, ctx, amount, member: discord.Member = None):
-        log_channel = self.bot.get_channel(int(os.getenv("MODERATION_LOGS_CHANNEL")))  # moderation-logs channel
-        emb = discord.Embed()
-        emb.set_author(name="Moderation",
-                       icon_url="https://cdn.discordapp.com/attachments/336577284322623499/683028692133216300/ac6e275e1f638f4e19af408d8440e1d1.png")
-        emb.set_footer(text=f'\t\t\t\t\t\t\tTimestamp: {ctx.message.created_at}')
-
-        # If specific memeber is not specified
-        if not member:
-            emb.add_field(name="Message (Clean)", value=f'<#{ctx.message.channel.id}>', inline=False)
-            emb.add_field(name="Amount", value=f'{amount}', inline=False)
-            emb.add_field(name="Moderator",
-                          value=f'{ctx.message.author}#{ctx.message.author.discriminator} ({ctx.message.author.id})',
-                          inline=False)
-            await ctx.message.channel.purge(limit=int(amount) + 1)
-            await log_channel.send(embed=emb)
-            return
-
-        # If specific member is specified
-        emb.add_field(name="Message (Clean)", value=f'<#{ctx.message.channel.id}>', inline=False)
-        emb.add_field(name="Amount", value=f'{amount}', inline=False)
-        emb.add_field(name="From", value=f'{member.mention} ({member.id})', inline=False)
-        emb.add_field(name="Moderator",
-                      value=f'{ctx.message.author}#{ctx.message.author.discriminator} ({ctx.message.author.id})',
-                      inline=False)
-
-        def is_member(message):
-            return message.author.id == member.id
-
-        await ctx.message.channel.purge(limit=int(amount) + 1, check=is_member)
-        await log_channel.send(embed=emb)
+    async def clean(self, ctx, amount: int, member: discord.Member = None):
+        _, emb, ch = prepare_action(ctx, '')
+        emb.add_field(name='Message (Clean)', value=ch.mention, inline=False)
+        emb.add_field(name="Moderator", value=ctx.message.author.mention, inline=False)
```

Comment:
I should probably move this to line 165

Reply:
If it makes sense, I don't have an issue with it.

Reply:
e16c1b4